ext/intl/converter/converter.c

```diff
@@ -376,25 +376,31 @@ static bool php_converter_set_encoding(php_converter_object *objval,
 			/* Should never happen */
 			actual_encoding = "(unknown)";
 		}
-		php_error_docref(NULL, E_WARNING, "Ambiguous encoding specified, using %s", actual_encoding);
+		char *msg;
+		spprintf(&msg, 0, "Ambigious encoding specified, using %s", actual_encoding);
+		intl_error_set(NULL, error, msg, 1);
+		efree(msg);
 	} else if (U_FAILURE(error)) {
 		if (objval) {
 			THROW_UFAILURE(objval, "ucnv_open", error);
 		} else {
-			php_error_docref(NULL, E_WARNING, "Error setting encoding: %d - %s", (int)error, u_errorName(error));
+			char *msg;
+			spprintf(&msg, 0, "Error setting encoding: %d - %s", (int)error, u_errorName(error));
+			intl_error_set(NULL, error, msg, 1);
```

Comment:
Should we pass `error` to the `code` parameter here? This is exposed via `intl_get_error_code()` so it can be useful.

In the `Ambigious encoding specified` case above, if we convert it back to a warning, we could still expose the error code by calling `intl_errors_set_code()`.

Reply:
well objval is null in that case, what do you suggest ?

Reply:
I was suggesting to pass `error` to the `code` parameter, but I'm noticing now that you are already doing exactly that, so please dismiss this comment :D